src/ASM/ASMs2DLag.C

```diff
@@ -682,18 +682,56 @@ bool ASMs2DLag::evalSolution (Matrix& sField, const Vector& locSol,
 
 
 bool ASMs2DLag::evalSolution (Matrix& sField, const Vector& locSol,
-                              const RealArray*, bool, int, int) const
+                              const RealArray* gpar, bool, int, int) const
 {
-  size_t nPoints = coord.size();
-  size_t nNodes = this->getNoNodes(-1);
-  size_t nComp = locSol.size() / nNodes;
-  if (nNodes < nPoints || nComp*nNodes != locSol.size())
-    return false;
+  if (!gpar) {
+    size_t nPoints = coord.size();
+    size_t nNodes = this->getNoNodes(-1);
+    size_t nComp = locSol.size() / nNodes;
+    if (nNodes < nPoints || nComp*nNodes != locSol.size())
+      return false;
+
+    sField.resize(nComp,nPoints);
+    const double* u = locSol.ptr();
+    for (size_t n = 1; n <= nPoints; n++, u += nComp)
+      sField.fillColumn(n,u);
+
+    return true;
+  }
+
+  size_t nNodes = gpar[0].size()*gpar[1].size();
+  size_t nComp = locSol.size() / this->getNoNodes();
+
+  sField.resize(nComp, nNodes);
+
+  FiniteElement fe(p1*p2);
```

Comment:
Don't see the need for the FiniteElement object here (no integrand involved).
Could just use single variables for N, xi and eta instead.

Reply:
I can..